### PR TITLE
specgen, cgroup2: check whether memory swap is enabled

### PR DESCRIPTION
**Pull request by giuseppe**
add a similar check to what we do on cgroup v1.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
Doesn't this fix an issue?

File: pkg/specgen/generate/validate.go [hunk 1]
```diff
@@ -1,32 +1,28 @@
...
 
 	if s.ResourceLimits == nil {
 		return warnings, nil
 	}
 
 	if s.ResourceLimits.Unified != nil {
-		if !cgroup2 {
-			return nil, errors.New("Cannot use --cgroup-conf without cgroup v2")
-		}
+		return nil, errors.New("Cannot use --cgroup-conf without cgroup v2")
```
Comment (mheon) [inline pkg/specgen/generate/validate.go:25]: Don't we still need the v2 check here?

Comment (mheon) [reply pkg/specgen/generate/validate.go:25]: Ah, I see, we have separate functions for v1 and v2

Comment (giuseppe) [reply pkg/specgen/generate/validate.go:25]: `verifyContainerResourcesCgroupV1` is called only on cgroup v1, so the extra check is superflous

**Comment by mheon** [comment]
LGTM

**Comment by rhatdan** [comment]
Fixes: https://github.com/containers/podman/issues/6365

**Comment by rhatdan** [comment]
@giuseppe Tests are not happy.

**Pull request merged**